### PR TITLE
[selectors-4] Allow logical combination pseudos wherever their arguments are themselves valid. #7085

### PR DESCRIPTION
**Pull request by fantasai**

### CODE REVIEW

File: selectors-4/Overview.bs [hunk 1]
```diff
@@ -988,6 +992,13 @@ Invalid Selectors and Error Handling</h3>
 <h2 id="logical-combination">
 Logical Combinations</h2>
 
+	Selector logical can be manipulated by
```
Comment (Loirooriol) [inline selectors-4/Overview.bs:995]: Should this be "Selector logic"? I don't understand an adjective here.

File: selectors-4/Overview.bs [hunk 2]
```diff
@@ -831,8 +829,14 @@ Pseudo-classing Pseudo-elements</h4>
 		whereas ''::first-line:hover'' only matches if the first line itself is hovered.
 	</div>
 
+	The [=logical combination pseudo-classes=]
+	'':not()'', '':is()'', and '':where()''
+	are also valid immediately following a [=pseudo-element=]
+	provided their arguments all adhere to the above restriction.
```
Comment (Loirooriol) [inline selectors-4/Overview.bs:835]: Does this mean that `::before:is(:hover, :first-child)` is completely invalid, or becomes `::before:is(:hover)`?

File: selectors-4/Overview.bs (hunk 1 above)
Comment (fantasai) [reply selectors-4/Overview.bs:995]: yes, that was a typo :) I'll fix it

File: selectors-4/Overview.bs (hunk 2 above)
Comment (fantasai) [reply selectors-4/Overview.bs:835]: Seems like the latter would be more consistent with standard behavior, so I'll update the PR.

**Pull request merged**